feat(prometheus): Add configuration option to hide warnings in UI

anglerfishlyy opened the pull request:
**What is this feature?**
This feature introduces a new configuration option in the Prometheus data source that allows users to hide warnings returned by Prometheus from being displayed in Grafana panels. When enabled, warnings are filtered out at the backend response parser level, providing a cleaner visualization experience.

**Why do we need this feature?**
Prometheus responses may include warnings (for example, query execution issues or deprecation notices). While useful in some cases, they can also clutter dashboards and distract end users who only care about the actual data. This option gives users control over whether to see these warnings or keep their panels clean.

**Who is this feature for?**

- Dashboard users who want to avoid UI clutter.

- Operators who don’t need end users exposed to Prometheus warnings.

- Anyone configuring Prometheus data sources who prefers a simplified panel experience.

**Which issue(s) does this PR fix?**
Fixes grafana/grafana-prometheus-datasource#143

**Special notes for your reviewer:**

Added hideWarnings flag in Prometheus data source settings.

- Updated parseResponse in response.go to strip warnings when enabled.

- Added UI toggle (Hide Prometheus warnings) under Advanced settings in the Prometheus data source configuration editor.

- Defaults to showing warnings (current behavior remains unchanged).

**Please check that:**

 [ ] The toggle works as expected in the data source configuration UI.

[ ] Backend filtering correctly removes warnings when enabled.

 [ ] Documentation is updated to include this new option.

Commit:
feat(prometheus): add datasource config to hide warnings in UI (#110480)

Commit:
docs(prometheus): add documentation for hide warnings option

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=110653) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=110653) before we can accept your contribution.<br/><hr/>**anglerfishlyy** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=110653) it.</sub>

anglerfishlyy committed:
Update prom.go

anglerfishlyy committed:
feat(prometheus): add configuration option to hide warnings in UI

anglerfishlyy committed:
feat(prometheus): add datasource option to hide warnings in UI

- Add  option in Prometheus datasource JSONData
- Pass  flag through QueryData execution to backend
- Prevent warnings from reaching frontend when enabled

anglerfishlyy committed:
Merge branch 'main' into feat/prometheus-hide-warnings

itsmylife commented on packages/grafana-prometheus/src/datasource.ts:
```diff
@@ -463,13 +468,13 @@ export class PrometheusDatasource
       return this.directAccessError();
     }
 
-    // Use incremental query only if enabled and no instant queries or no $__range variables
```
Please don't delete this comment

itsmylife reviewed (changes requested):
Thanks for the contribution. There are missing parts here and there. I'd suggest you to not handle the warnings in the frontend-side. I'd rather do it in backend. While getting a new instance of QueryData we can set hideWarnings config and use it in converter to opt-out the warnings.
